libp2p.nimble

```diff
@@ -109,52 +99,23 @@ task examples_build, "Build the samples":
     buildSample("tutorial_3_protobuf", true)
     buildSample("tutorial_4_gossipsub", true)
     buildSample("tutorial_5_discovery", true)
-    # Nico doesn't work in 1.2
-    exec "nimble install -y nimpng@#HEAD" # this is to fix broken build on 1.7.3, remove it when nimpng version 0.3.2 or later is released
-    exec "nimble install -y nico"
     buildSample("tutorial_6_game", false, "--styleCheck:off")
 
 # pin system
 # while nimble lockfile
 # isn't available
 
-const PinFile = ".pinned"
-task pin, "Create a lockfile":
-  # pinner.nim was originally here
-  # but you can't read output from
-  # a command in a nimscript
-  exec "nim c -r tools/pinner.nim"
-
-import sequtils
-import os
-task install_pinned, "Reads the lockfile":
```

Comment:
As discussed on discord, I'd rather keep this to have compat with old nimble, and maybe do
```nim
task install_pinned "Reads the lockfile":
  when nimbleIsCompatibleWithNewSystem:
    exec "nimble install -dy" # or whatever
  else:
    # the old code
```